packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -151,11 +151,51 @@ abstract class GestureRecognizer extends GestureArenaMember with DiagnosticableT
 /// simultaneous touches to each result in a separate tap.
 abstract class OneSequenceGestureRecognizer extends GestureRecognizer {
   /// Initialize the object.
-  OneSequenceGestureRecognizer({ Object debugOwner }) : super(debugOwner: debugOwner);
+  ///
+  /// {@template flutter.gestures.oneSequenceGestureRecognizer.kind}
+  /// It's possible to limit this recognizer to a specific [PointerDeviceKind]
+  /// by providing the optional [kind] argument. If [kind] is null,
+  /// the recognizer will accept pointer events from all device kinds.
+  /// {@endtemplate}
+  OneSequenceGestureRecognizer({
+    Object debugOwner,
+    @required PointerDeviceKind kind,
+  }) :  _kind = kind,
+        super(debugOwner: debugOwner);
+
+  /// The kind of device that's allowed to be recognized. If null, events from
+  /// all device kinds will be tracked and recognized.
+  final PointerDeviceKind _kind;
 
   final Map<int, GestureArenaEntry> _entries = <int, GestureArenaEntry>{};
   final Set<int> _trackedPointers = HashSet<int>();
 
+  @override
+  void addPointer(PointerDownEvent event) {
+    if (isPointerAllowed(event)) {
+      addAllowedPointer(event);
+    } else {
+      resolve(GestureDisposition.rejected);
+    }
+  }
+
+  /// Registers a new pointer that's been checked to be allowed by this gesture
+  /// recognizer.
+  ///
+  /// Subclasses of [OneSequenceGestureRecognizer] are supposed to override this
+  /// method instead of [addPointer] because [addPointer] will be called for each
+  /// pointer being added while [addAllowedPointer] is only called for pointers
+  /// that are allowed by this recognizer.
+  @protected
+  void addAllowedPointer(PointerDownEvent event) { }
```

Comment:
double tap isn't a OneSequenceGestureRecognizer

Reply:
Ok, changed it so that all subclasses of `GestureRecognizer` can filter based on device kind.